### PR TITLE
Fixes to and resolution of root cause failures in core animation tests

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: packages/core/test/animation/animation_integration_spec.ts [hunk 1]
```diff
@@ -3373,7 +3373,7 @@ const DEFAULT_COMPONENT_ID = '1';
              expect(getLog().length).toEqual(1);
            });
 
-        fixmeIvy('unknown').it(
+        fixmeIvy('unknown').fit(
```
Comment (kara) [inline packages/core/test/animation/animation_integration_spec.ts]: Remove fit?

**Review by mhevery — APPROVED** [review]

File: packages/compiler/src/render3/view/template.ts [hunk 2]
```diff
@@ -688,10 +688,10 @@ export class TemplateDefinitionBuilder implements t.Visitor<void>, LocalResolver
       const instruction = mapBindingToInstruction(input.type);
       if (input.type === BindingType.Animation) {
         const value = input.value.visit(this._valueConverter);
-        // setProperty without a value doesn't make any sense
-        if (value.name || value.value) {
-          const bindingName = prepareSyntheticPropertyName(input.name);
+        const hasValue = value && (value instanceof LiteralPrimitive) ? !!value.value : true;
```
Comment (kara) [inline packages/compiler/src/render3/view/template.ts]: Can you add a comment here to explain this check?

**Review by kara — CHANGES_REQUESTED** [review]

**Review by kara — APPROVED** [review]
LGTM

**Comment by matsko** [comment]
https://tap.corp.google.com/ui#id=OCL:228588746:BASE:228598338:1547075921206:5babac25

**Comment by AndrewKushnir** [comment]
@matsko new g3 presubmit (the old one has conflicts, since the g3 sync CL just landed):
http://test/OCL:228600272:BASE:228600324:1547076679434:fdf70e06

**Comment by matsko** [comment]
Thank you @AndrewKushnir

**Pull request closed without merging**